core/src/main/scala/org/apache/spark/sql/execution/ExistingPlans.scala

```diff
@@ -186,7 +211,7 @@ private[sql] final case class ZipPartitionScan(basePlan: SparkPlan with CodegenS
     val row = ctx.freshName("row")
     val columnsInputEval = Option(otherPlan).getOrElse(basePlan).output.zipWithIndex.map { case
       (ref, ordinal) =>
-      val baseIndex = Option(otherPlan).fold(relation.schema.fieldIndex(ref.name))(_ => ordinal)
+      val baseIndex = Option(otherPlan).fold(relation.get.schema.fieldIndex(ref.name))(_ => ordinal)
```

Comment:
Same as above, Option.fold behaves differently than Col.fold, @vivekwiz find, I kind of liked this oneliner :)